compiler/backend/src/org/jetbrains/kotlin/codegen/inline/defaultMethodUtil.kt

```diff
@@ -202,3 +179,68 @@ private fun defaultLambdaFakeCallStub(args: Array<Type>, lambdaOffset: Int): Met
         false
     )
 }
+
+fun loadDefaultLambdaBody(classBytes: ByteArray, classType: Type, isPropertyReference: Boolean): SMAPAndMethodNode {
+    // In general we can't know what the correct unboxed `invoke` is, and what Kotlin types its arguments have,
+    // as the type of this object may be any subtype of the parameter's type. All we know is that Function<N>
+    // has to have a `invoke` that takes `Object`s and returns an `Object`; everything else needs to be figured
+    // out from its contents. TODO: for > 22 arguments, the only argument is an array. `MethodInliner` can't do that.
+    val invokeName = if (isPropertyReference) OperatorNameConventions.GET.asString() else OperatorNameConventions.INVOKE.asString()
+    val invokeNode = getMethodNode(classBytes, classType) {
+        it.name == invokeName && it.returnType == AsmTypes.OBJECT_TYPE && it.argumentTypes.all { arg -> arg == AsmTypes.OBJECT_TYPE }
+    } ?: error("can't find erased invoke '$invokeName(Object...): Object' in default lambda '${classType.internalName}'")
+    return if (invokeNode.node.access.and(Opcodes.ACC_BRIDGE) == 0)
+        invokeNode
+    else
+        invokeNode.node.inlineBridge(classBytes, classType)
+}
+
+private fun MethodNode.inlineBridge(classBytes: ByteArray, classType: Type): SMAPAndMethodNode {
+    // If the erased invoke is a bridge, we need to locate the unboxed invoke and inline it. As mentioned above,
+    // we don't know what the Kotlin types of its arguments/returned value are, so we can't generate our own
+    // boxing/unboxing code; luckily, the bridge already has that.
+    val invokeInsn = instructions.singleOrNull { it is MethodInsnNode && it.owner == classType.internalName } as MethodInsnNode?
+        ?: error("no single invoke of method on this in '${name}${desc}' of default lambda '${classType.internalName}'")
+    val targetMethod = Method(invokeInsn.name, invokeInsn.desc)
+    val target = getMethodNode(classBytes, classType, targetMethod)
+        ?: error("can't find non-bridge invoke '$targetMethod' in default lambda '${classType.internalName}")
+
+    // Store unboxed/casted arguments in the correct variable slots
+    val targetArgs = targetMethod.argumentTypes
+    val targetArgsSize = targetArgs.sumOf { it.size } + if (target.node.access.and(Opcodes.ACC_STATIC) == 0) 1 else 0
+    var offset = targetArgsSize
+    for (type in targetArgs.reversed()) {
+        offset -= type.size
+        instructions.insertBefore(invokeInsn, VarInsnNode(type.getOpcode(Opcodes.ISTORE), offset))
+    }
+    if (target.node.access.and(Opcodes.ACC_STATIC) == 0) {
+        instructions.insertBefore(invokeInsn, InsnNode(Opcodes.POP)) // this
```

Comment:
Not sure, but look like that on static call `this`  shouldn't be generated

Reply:
That's the point of the `if`, yes...though there are no static `invoke`s (at least right now) anyway.

Reply:
yes, all of them are virtual because of bridges